async_simple/coro/test/LazyTest.cpp

```diff
@@ -1104,6 +1104,104 @@ TEST_F(LazyTest, testCollectAllVariadic) {
     syncAwait(testCollectAllPara().via(&e1));
 }
 
+TEST_F(LazyTest, testCollectAnyWithCallbackVariadic) {
+    auto test0 = []() -> Lazy<Unit> { co_return Unit{}; };
+    auto test1 = []() -> Lazy<int> { co_return 42; };
+    auto test2 = [](int val) -> Lazy<std::string> {
+        co_return std::to_string(val);
+    };
+
+    size_t index = syncAwait(collectAny(std::pair{test0(), [](auto val) {}}));
+
+    index = syncAwait(collectAny(
+        std::pair{test1(), [](auto val) { EXPECT_EQ(42, val.value()); }}));
+
+    index = syncAwait(collectAny(
+        std::pair{test2(42), [](auto str) { EXPECT_EQ("42", str.value()); }}));
+    EXPECT_EQ(index, 0);
+
+    int call_count = 0;
+    index =
+        syncAwait(collectAny(std::pair{test0(), [&](auto) { call_count++; }},
+                             std::pair{test1(),
+                                       [&](auto val) {
+                                           call_count++;
+                                           EXPECT_EQ(val.value(), 42);
+                                       }},
+                             std::pair{test2(42), [&](auto val) {
+                                           call_count++;
+                                           EXPECT_EQ("42", val.value());
+                                       }}));
+    EXPECT_EQ(1, call_count);
+
+    executors::SimpleExecutor e1(4);
+    auto test3 = []() -> Lazy<int> { co_return 42; };
+    auto test4 = [](int val) -> Lazy<int> { co_return val; };
+
+    int test_value = 0;
+    index = syncAwait(collectAny(std::pair{test3().via(&e1),
+                                           [&](auto val) {
+                                               test_value = val.value();
+                                               EXPECT_EQ(42, test_value);
+                                           }},
+                                 std::pair{test4(41).via(&e1), [&](auto val) {
+                                               test_value = val.value();
+                                               EXPECT_EQ(41, test_value);
+                                           }}));
```

Comment:
It will be better if we can avoid writing `std::pair` and try to use `{, }`. It may be possible to construct a pair from the initializer list.

Reply:
I think maybe it's impossible: https://stackoverflow.com/questions/65071836/template-argument-deduction-for-implicit-pair

can't use std::initializer_list to avoid std::pair with brace initialize.

another reason, can't use std::initializer_list because it requires types are the same, not variadic template.

Reply:
Got it.